### PR TITLE
Update doc around Resteasy GZIP support

### LINKED ISSUE #9671: GZip filter doesn't work
**Describe the bug**
I followed the steps in https://quarkus.io/guides/rest-json#gzip-support but the filter doesn't seem to be activated.

**Expected behavior**
Content will be compressed as GZIP 

**Actual behavior**
No GZIP compression

**To Reproduce**
1. Create any endpoint with code.quarkus.io
2. Add `quarkus.resteasy.gzip.enabled=true` to application.properties
3. Run `http localhost:8080/hello Content-Encoding:gzip Accept-Encoding:gzip`

**Configuration**
```properties
quarkus.resteasy.gzip.enabled=true
```
**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Fedora 32
 - Output of `java -version`: OpenJDK 11
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.5.0.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Maven 3.6.3 

**Additional context**

It looks like the `Content-Encoding` and `Accept-Encoding` headers are not propagated to the internal Vert.x Request, thus making `org.jboss.resteasy.plugins.interceptors.GZIPEncodingInterceptor` ignore the request to GZIP the output

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #9671

### CODE REVIEW

File: extensions/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/gzip/GZipResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    static QuarkusUnitTest test = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(GZipResource.class))
+            .overrideConfigKey("quarkus.resteasy.gzip.enabled", "true");
+
+    @Test
+    public void testGzipEndpoint() {
+        RestAssured.given().get("/gzip").then().statusCode(200)
+                .header("content-encoding", "gzip")
+                .header("content-length", Matchers.not(Matchers.equalTo(Integer.toString(GZipResource.BODY.length()))))
```
Comment (gastaldi) [inline extensions/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/gzip/GZipResourceTest.java:24]: Static importing `Matches` makes the code cleaner

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by BastianSperrhacke-Otto** [comment]
I tried the ```quarkus.http.enable-compression=true``` Property and this works completely as expected. Thank you!!

File: extensions/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/gzip/GZipResourceTest.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/gzip/GZipResourceTest.java:24]: @gastaldi you can always push that into Stuart's PR :)

**Review by gsmet — APPROVED** [review]

**Pull request merged**